client/v3/lease.go

```diff
@@ -348,7 +353,7 @@ func (l *lessor) keepAliveCtxCloser(ctx context.Context, id LeaseID, donec <-cha
 
 	// close channel and remove context if still associated with keep alive
 	for i, c := range ka.ctxs {
-		if c == ctx {
+		if val := c.Value(ctx); val != nil {
```

Comment:
I am confused about this change. 

Have you considered https://github.com/etcd-io/etcd/pull/18893#issuecomment-2486198060 and also https://github.com/etcd-io/etcd/pull/18893#issuecomment-2486156823?